### DISCUSSION TITLE
Possible to port docker image to work on ARM (Synology)

### DISCUSSION THREAD

**Discussion by bmcgowan32** [sktaylortrash/TCPLightingWebInterface-MQTT-Docker#6] [General]
Hey!  Ive been rocking the docker version for a year now and your container for TCP lights is one of my best automation add-ons.  Im starting to migrate services over from my Ubunutu server to Container Manager on my Synology NAS, but Im having trouble as I think your container is platformed for Linux/AMD64 and it wont run on the ARM64 hardware in my NAS.  Is there a chance you'd consider building for an ARM platform, or a cross platform container image?  Or maybe I can create a fork for ARM...  

THANK YOU SO MUCH FOR THIS PROJECT.  I would've junked all my TCP lights if you hadnt saved them with this great project

**Comment by sktaylortrash** [comment]
I've taken a quick look, and if the base php:7.4-apache image has an ARM64 build, this probably won't be too hard to do. 

I don't think I can test it, as I don't have anything with an ARM64 CPU.  Unless I have a PI 4 floating around in a drawer (which I might)

I likely won't be able to dig into it until the weekend. But will definitely take a look for you.

**Comment by bmcgowan32** [comment]
That would be amazing!  Im happy to help with testing?  I really apprecaite that you continue to support this project.

**Comment by sktaylortrash** [comment]
Ok had a few minutes to mess with this.  I have no idea if it will work or not, but there's now a multiple architecture build available for testing you'll want to change the image tag in docker-compose.yml from latest to multiarchbeta

**Reply by bmcgowan32** [reply]
Awesome!  Thanks so much!  The new build loads and runs successfully, but it looks llike Im getting an error with the MQTT conection.  Possibly its the single quotes in the string?

<img width="1189" alt="Screenshot 2024-08-07 at 8 33 21 AM" src="https://github.com/user-attachments/assets/0e20b83c-50e9-49c2-b01d-91e55a3bcd4f">

**Reply by bmcgowan32** [reply]
Thanks for looking at it.  There is NO RUSH.  I appreciate you spending your personal time on this.  Lets hope Im not the only person who would like to run the container on a NAS?

**Reply by bmcgowan32** [reply]
Disregard that log.  It turns out there was a permissions issue with the token file on the NAS.  I got that fixed, and re-ran it, and the UI lodaed!  But.. When I tried to control a light, I got this error and the container crashed:

<img width="1205" alt="Screenshot 2024-08-07 at 8 57 24 AM" src="https://github.com/user-attachments/assets/08e1ad80-7b45-40f1-b0ae-e66537c0e8a1">

**Reply by bmcgowan32** [reply]
Im not ruling out a networking issue in the container yet.  The web UI loads, pulls in the current status of the lights from the bridge, and runs until I try to change a light level/status.  When I do, the physical light does change state, so the HTTPS call to the bridge is working, but when the MQTT call to the broker fires, the container crashes.  Im a little suspicious that the MQTT call is trying to use the internal IP of the container...  Ill dig a bit more
